frame/treasury/src/lib.rs

```diff
@@ -187,10 +187,7 @@ decl_storage! {
 			let account_id = <Module<T, I>>::account_id();
 			let min = T::Currency::minimum_balance();
 			if T::Currency::free_balance(&account_id) < min {
-				let _ = T::Currency::make_free_balance_be(
-					&account_id,
-					min,
-				);
+				let _ = T::Currency::make_free_balance_be(&account_id, min);
```

Comment:
Shouldn't we replace the `_` with a debug assert?

Reply:
I had some problems with the `make_free_balance_be` functions. See https://github.com/paritytech/substrate/issues/8106#issuecomment-814224971

Reply:
You could always match over a reference to prevent consuming the value. In this case we would expect a positive imbalance.

Reply:
Ah, I see. There are many places to check `make_free_balance_be` I'll go over the codebase again and try to fix them.

Reply:
One pattern common would be to add one method per variant to an enum (`SignedImbalance` in this case) which return an optionals. This avoids the `match` at the call side. For example:
```rust
fn as_positive(&self) -> Option<&Self::PositiveImbalance>
```

Reply:
Or using the `assert_matches` crate which we already have as dependency?

Reply:
That is fine, too.

Reply:
make_free_balance doesnt need a check right?

There is no "Failure" situation. It always does what it is meant to do.

Reply:
Yeah. It does not return a result. But why was there a `let _ =` in the first place?

Reply:
To create a scoped object that has a custom drop-handler?

Reply:
Should not be needed. Just drop it or leave it. This is not one of those that you need to fix.

Reply:
Ok, then I think we're done